### PR TITLE
update hubot version

### PR DESCRIPTION
**Pull request by gorkem**
Updates hubot, hubot-help and matteruser to new version. Also remove the intermediary classes and instead uses the typings. Adds typings for github-url-to-object.

### CODE REVIEW

**Comment by gorkem** [comment]
@adietish Can you do a review

**Comment by adietish** [comment]
I think I hit the following error: https://stackoverflow.com/questions/51364313/hubot-mattermost-may-be-incompatible-with-hubot3-0-1

When reporting a standup I see the reminder service hitting the following error at https://github.com/gorkem/penfold/blob/master/src/reminder/service.ts#L57
```
Exception has occurred: TypeError
TypeError: ReminderService.robot.getUserForId is not a function
  at Report.findOne.then.result (/Users/adietish/Documents/jboss-workspaces/penfold/src/reminder/service.ts:57:50)
  at newTickHandler (/Users/adietish/Documents/jboss-workspaces/penfold/node_modules/mpromise/lib/promise.js:234:18)
  at process._tickCallback (internal/process/next_tick.js:61:11)
```

**Comment by adietish** [comment]
it looks like there are a few more tasks to be tackled for the upgrade to work. When trying to tackle them I stumbled upon this bug we should fix first: https://github.com/gorkem/penfold/issues/22
@gorkem can you please review https://github.com/gorkem/penfold/issues/22 which fixes it?

File: package.json [hunk 1]
```diff
@@ -8,9 +8,9 @@
     "@octokit/rest": "^15.8.2",
     "agenda": "^0.9.1",
     "github-url-to-object": "4.0.4",
-    "hubot": "^2.19.0",
-    "hubot-help": "^0.2.2",
-    "hubot-matteruser": "^4.4.0",
+    "hubot": "^3.1.1",
```
Comment (adietish) [inline package.json]: dependabot suggested this upgrade in PR (#42)[https://github.com/gorkem/penfold/pull/42/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R12]

File: package.json [hunk 2]
```diff
@@ -8,9 +8,9 @@
...
+    "hubot-help": "^1.0.1",
```
Comment (adietish) [inline package.json]: dependabot is suggesting this in PR [#40](https://github.com/gorkem/penfold/pull/40/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R13)

File: package.json [hunk 3]
```diff
@@ -8,9 +8,9 @@
...
+    "hubot-matteruser": "^5.2.0",
```
Comment (adietish) [inline package.json]: this was not suggested by dependabot.

**Comment by adietish** [comment]
@gorkem I'd suggest that we drop the different dependabot PRs in favor of this one?

File: package.json [hunk 4]
```diff
@@ -22,18 +22,19 @@
     "node": "0.10.x"
   },
   "devDependencies": {
-    "typescript": "^2.4.1",
-    "@types/node": "7.0.12",
-    "@types/mocha": "^2.2.40",
     "@types/chai": "3.4.35",
+    "@types/github-url-to-object": "^4.0.0",
+    "@types/mocha": "^2.2.40",
```
Comment (adietish) [inline package.json]: dependabot suggested an uprade to 5.2.5 in PR [#33] (https://github.com/gorkem/penfold/pull/33/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R30)

**Pull request merged**